GVRf/Framework/jni/engine/renderer/renderer.cpp

```diff
@@ -59,17 +65,46 @@ void Renderer::renderCamera(std::shared_ptr<Scene> scene,
             std::shared_ptr<RenderData> render_data = (*it)->render_data();
             if (render_data != 0) {
                 if (render_data->material() != 0) {
-                    render_data_vector.push_back(render_data);
+                    // Check for frustum culling flag
+                    if(scene->get_frustum_culling())
+                    {
+                        // Frustum  Culling
+                        std::shared_ptr<Mesh> currentMesh = render_data->mesh();
+                        float* bounding_box_info = currentMesh->getBoundingBoxInfo();
```

Comment:
it looks like getBoundingBoxInfo() iterates through all vertices of the Mesh each time it is called.  Also getBoundingBoxInfo() has a lot of redundant code with getBoundingBox().  Seems to me that when the Mesh is created, we could call getBoundingBox() to create the bounding box, store any extra info needed for culling, then just query that stored info when culling.  If there are any changes to the mesh, that would dirty the bounding box and force it to be recomputed.